modules/file-system/pre-existing-network-storage/scripts/mount.sh

```diff
@@ -19,18 +19,6 @@ LOCAL_MOUNT=$3
 FS_TYPE=$4
 MOUNT_OPTIONS=$5
 
-if ! findmnt --source "${SERVER_IP}":"${REMOTE_MOUNT_WITH_SLASH}" --target "${LOCAL_MOUNT}" &>/dev/null; then
-	echo "Mounting --source ${SERVER_IP}:${REMOTE_MOUNT_WITH_SLASH} --target ${LOCAL_MOUNT}"
-	mkdir -p "${LOCAL_MOUNT}"
-	if [ -z "${MOUNT_OPTIONS}" ]; then
-		mount -t "${FS_TYPE}" "${SERVER_IP}":"${REMOTE_MOUNT_WITH_SLASH}" "${LOCAL_MOUNT}"
-	else
-		mount -t "${FS_TYPE}" -o "${MOUNT_OPTIONS}" "${SERVER_IP}":"${REMOTE_MOUNT_WITH_SLASH}" "${LOCAL_MOUNT}"
-	fi
-else
-	echo "Skipping mounting source: ${SERVER_IP}:${REMOTE_MOUNT_WITH_SLASH}, already mounted to target:${LOCAL_MOUNT}"
-fi
-
 MOUNT_IDENTIFIER="${SERVER_IP}:${REMOTE_MOUNT_WITH_SLASH}[[:space:]]${LOCAL_MOUNT}"
 if ! grep -q "${MOUNT_IDENTIFIER}" /etc/fstab; then
```

Comment:
I believe here you should look for only ${SERVER_IP}:${REMOTE_MOUNT_WITH_SLASH} and not the local mount. The reason is that you may want to mount the same file system in multiple places.. right? 
However, looking for only the SERVER_IP will not be sufficient either, because multiple remote mounts may exist under the same ip.